rmw_fastrtps_cpp/src/rmw_publisher.cpp

```diff
@@ -128,8 +128,13 @@ rmw_create_publisher(
     goto fail;
   }
 
-  info->publisher_ = Domain::createPublisher(participant, publisherParam, nullptr);
+  info->listener_ = new PubListener(info);
+  if (!info->listener_) {
```

Comment:
Unless you use the `nothrow` version of `new`, `new` will throw an exception (`std::bad_alloc`), so this should never be reached when it's `nullptr`.